sql-common/my_time.c

```diff
@@ -336,12 +339,25 @@ str_to_datetime(const char *str, uint length, MYSQL_TIME *l_time,
     pos++;
     skip_digits(&pos, end); // ignore the return value
   }
+  if (pos < end && *pos == '-' && digits >= 12) /* YYYYYMMDDHHMMSShhmmss.uuuuuu-HH:MM is supported too */
+  {
+    pos+=6; // Skip 6 places as defined in RFC3339 reserved for local offset
+  }
+  else if (pos < end && *pos == '+' && digits >= 12) /* YYYYYMMDDHHMMSShhmmss.uuuuuu+HH:MM is supported too */
+  {
+    pos+=6; // Skip 6 places as defined in RFC3339 reserved for local offset
```

Comment:
Ignoring the local offset rather than merging it into the time seems to be a mistake to me. Recommend parsing the local offset and making it part of the time.